wordpress-mobile/WordPress-Android
# Issue/16122 add blogging prompt option to reminders

Fixes #16122

This PR adds a blogging prompt Feature Flag, and adds a blogging prompt specific UI to cadence settings when the feature flag is on.

[![Image from Gyazo](https://i.gyazo.com/13171796b7138e676d4e3ad0e881f0af.png)](https://gyazo.com/13171796b7138e676d4e3ad0e881f0af)

[![Image from Gyazo](https://i.gyazo.com/807f10744a667a13e583a35ae97741e3.png)](https://gyazo.com/807f10744a667a13e583a35ae97741e3)

To test:
- You want to compare the changes alongside a current version from the trunk.
- Perform a clean install of wasabi build variant.
- Navigate to App Settings -> Debug settings and enable `blogging_prompts_remote_field` option.
- Navigate back to My Site screen.
- Tap on + button and crate a new new post. After post is published you should see the blogging reminder prologue. (will appear only on first publish)
- Confirm that the copy in the prologue matches provided screenshots/design.
- Proceed to blogging prompts cadence settings.
- Select a day, and confirm that the "Include prompt" switch is present, and operates correctly.
- Confirm that the primary button label says "Set reminders" and not "Notify me" (it will say "Update" if you already set reminder before).
- Navigate to Site Settings and confirm that `Blogging reminders` now reads `Reminders and prompts`.
- Tap on the `Reminders and prompts` and confirm that prologue copy is same as prologue after posting.

## Regression Notes
1. Potential unintended areas of impact
- Wrong copy is visible when FF is off.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and new automated tests.

4. What automated tests I added (or what prevented me from doing so)
Tests for ff behavior.

PR submission checklist:
